gms/versioned_value.cc

```diff
@@ -58,4 +58,45 @@ versioned_value versioned_value::factory::network_version() {
     return versioned_value(format("{}", netw::messaging_service::current_version));
 }
 
+sstring versioned_value::make_full_token_string(const std::unordered_set<dht::token>& tokens) {
+    return ::join(";", tokens | boost::adaptors::transformed([] (const dht::token& t) {
+        return t.to_sstring(); })
+    );
+}
+
+sstring versioned_value::make_token_string(const std::unordered_set<dht::token>& tokens) {
+    if (tokens.empty()) {
+        return "";
+    }
+    return tokens.begin()->to_sstring();
```

Comment:
Nit: maybe 'return tokens.empty() ? "" : tokens.begin()->to_sstring();'

Reply:
It's code movement, so it's better to leave it as is in this patch - may be fixed as a follow-up later

Reply:
Yes. I agree. Move then further refactor.